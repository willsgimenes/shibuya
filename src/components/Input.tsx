import React, { FC, InputHTMLAttributes } from 'react';
import styled from 'styled-components';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  innerRef: any;
  disabled?: boolean;
  label: string;
}

const StyledInput = styled.input.attrs(
  ({ disabled }: { disabled: boolean }) => ({
    className: `p-2 border rounded-lg placeholder-yellow-800 ${disabled &&
      'border-gray-100 text-gray-400'} `,
  })
)``;

export const Input: FC<InputProps> = ({
  value,
  onChange,
  innerRef,
  disabled = false,
  label,
  ...props
}) => {
  return (
    <fieldset>
      <label htmlFor={label}>{label}</label>
      <StyledInput
        id={label}
        value={value}
        type="text"
        onChange={onChange}
        ref={innerRef}
        disabled={disabled}
        {...props}
      />
    </fieldset>
  );
};
