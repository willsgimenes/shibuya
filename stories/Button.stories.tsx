import React from 'react';
import { Meta, Story } from "@storybook/react";
import { Button, ButtonProps } from '../src/components/Button';

const meta: Meta = {
    title: 'Button',
    component: Button,
    argTypes: {
        text: {
            control: {
                type: 'text'
            }
        }
    },
    parameters: {
        controls: { expanded: true }
    }
}

export default meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const PrimaryButton = Template.bind({});
PrimaryButton.args = { text: 'Button', variant: 'primary' };

export const SecondaryButton = Template.bind({});
SecondaryButton.args = { text: 'Button', variant: 'secondary' }

export const TertitaryButton = Template.bind({});
TertitaryButton.args = { text: 'Button', variant: 'tertiary' }
