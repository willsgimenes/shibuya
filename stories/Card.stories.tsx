import React from "react";
import { Meta, Story } from "@storybook/react";
import { Card, CardProps } from "../src/components/Card";

const meta: Meta = {
    title: 'Card',
    component: Card,
}

export default meta;

const content = <div className='p-4'><p>渋谷区</p></div>

const Template: Story<CardProps> = (args) => <Card {...args} />

export const DefaultCard = Template.bind({});

DefaultCard.args = { children: content }
