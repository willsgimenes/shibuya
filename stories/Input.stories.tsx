import React, { useState, useRef } from 'react';
import { Meta, Story } from "@storybook/react";
import { Input, InputProps } from '../src/components/Input';

const meta: Meta = {
    title: 'Input',
    component: Input,
    argTypes: {
        value: {
            control: {
                type: 'text'
            }
        }
    },
    parameters: {
        controls: { expdanded: true }
    }
}

export default meta;

const Template: Story<InputProps> = (args) => React.createElement(() => {
    const [state, setState] = useState(args.value)
    const inputRef = useRef();

    return <Input
        innerRef={inputRef}
        value={state}
        label={args.label}
        onChange={({ target: { value } }) => setState(value)}
        placeholder={args.placeholder}
        disabled={args.disabled}
    />
})

export const DefaultInput = Template.bind({});
DefaultInput.args = { value: 'Yokohama Blues', placeholder: 'Insert text', disabled: false, label: 'song' }
