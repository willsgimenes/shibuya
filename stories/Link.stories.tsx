import React from "react";
import { Meta, Story } from "@storybook/react";
import { Link as SCLink, LinkProps } from '../src/components/Link';

const meta: Meta = {
    title: 'Link',
    component: SCLink,
}

export default meta;

const Template: Story<LinkProps> = (args) => (
    <SCLink {...args}>{args.children}</SCLink>
)

export const Link = Template.bind({});
Link.args = {
    url: "https://open.spotify.com/track/1EJOWPgFH4ThdIFdvbKA2B?si=Zdo4k4UbRnShLeKTHtDr6Q",
    external: true,
    children: 'Listen Yokohama blues',
}