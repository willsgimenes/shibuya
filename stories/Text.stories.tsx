import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Text, TextProps } from "../src/components/Text";

const meta: Meta = {
    title: 'Typography',
    component: Text,
}

export default meta;

const Template: Story<TextProps> = () => <div>
    <Text variant="h1">Heading 1</Text>
    <Text variant="h2">Heading 2</Text>
    <Text variant="h3">Heading 3</Text>
    <Text variant="h4">Heading 4</Text>
    <Text variant="h5">Heading 5</Text>
    <Text variant="p">texto</Text>
    <Text variant='small'>small text</Text>
</div>

export const DefaultTypography = Template.bind({})