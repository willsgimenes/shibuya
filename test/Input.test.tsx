import * as React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { DefaultInput } from '../stories/Input.stories';

describe('Input', () => {
  it('should render input without crashing', () => {
    const ref = {};
    const { getByPlaceholderText } = render(
      <DefaultInput label="song" innerRef={ref} placeholder="Insert text" />
    );

    const element = getByPlaceholderText('Insert text');
    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
  });

  it('should change value', () => {
    const ref = {};
    const mockedChange = jest.fn();

    const { getByPlaceholderText } = render(
      <DefaultInput
        label="song"
        innerRef={ref}
        placeholder="Insert text"
        onChange={mockedChange}
        value=""
      />
    );

    const element = getByPlaceholderText('Insert text');

    fireEvent.focus(element);
    fireEvent.change(element, { target: { value: 'silent' } });
    fireEvent.blur(element);
    expect(screen.getByDisplayValue('silent')).toBeInTheDocument();
  });

  it('should have disabled state', () => {
    const ref = {};
    const { getByPlaceholderText } = render(
      <DefaultInput
        label="song"
        innerRef={ref}
        placeholder="Insert text"
        value=""
        disabled
      />
    );

    const element = getByPlaceholderText('Insert text');
    expect(element).toBeDisabled();
  });
});
