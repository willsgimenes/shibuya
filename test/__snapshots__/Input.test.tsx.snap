// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render input without crashing 1`] = `
<input
  class="sc-bdfBwQ p-2 border rounded-lg placeholder-yellow-800 false "
  id="song"
  placeholder="Insert text"
  type="text"
  value=""
/>
`;
